### ISSUE TITLE
Duplicate decleration Package[redis] when both include redis and redis::sentinel

### ISSUE THREAD

**Issue by raiblue** [arioch/puppet-redis#36]
Hello,
I came across a problem when i used this module for managing both redis and sentinel on same node. On RHEL, both needs redis package, so `$::redis::package_name` in install.pp and `$package_name` in sentinel.pp is same. This results in duplicate decleration:

install.pp:
```
  ensure_resource('package', $::redis::package_name, {
    'ensure' => $::redis::package_ensure
  })
```
sentinel.pp:
```
  ensure_resource('package', $package_name, {
    'ensure' => $package_ensure
  })
```
I worked around by adding `unless defined` to them like,
install.pp:
```
  unless defined(Package["$::redis::package_name"]) {
  ensure_resource('package', $::redis::package_name, {
    'ensure' => $::redis::package_ensure
  })
  }
```

sentinel.pp:
```
  unless defined(Package["$package_name"]) {
  ensure_resource('package', $package_name, {
    'ensure' => $package_ensure
  })
  }
```

I think applying redis and sentinel to same node is a common behavior so this can be fixed.
Thanks for the nice module, by the way :)

**Comment by arioch** [comment]
Hi,

Thanks for reporting and investigating too.
Could you wrap this in a pull request? This way you get to have the proper credits for your fix.

**Comment by raiblue** [comment]
Hi again,
Sorry for the mess in pull requests, I don't know much about Github. I closed the first pr i did and opened a new one with just one commit.

**Comment by arioch** [comment]
No worries.
To me the easiest way of doing this is:
```
$ git rebase -i HEAD~2
replace 'pick' with 'f'
$ git push origin +master
```
This will automagically update the PR as well.
If you have new features you want to add to a PR you can just push those changes too.

**Comment by arioch** [comment]
I'll go ahead and close this one as it's already merged (963fc0cc3769c5ccbb7e1043699be1425f5e5b2f) via your other PR.
